nncf/experimental/torch/fx/transformations.py

```diff
@@ -528,6 +560,188 @@ def fuse_conv_bn(model: torch.fx.GraphModule) -> None:
+
+    return pattern_per_channel, replacement_graph_per_channel
+
+
+def _get_pattern_replacement_per_tensor() -> None:
+    """
+    Returns the patter and replacement function for the subgraph rewriter to 
+    match and replace for per_tensor quantization
+    """
+
+    def pattern_per_tensor(weight, scale, zero_point, low, high, dtype):
+        quantized = torch.ops.quantized_decomposed.quantize_per_tensor.default(
+            weight, scale, zero_point, low, high, dtype
+        )
+        dequantized = torch.ops.quantized_decomposed.dequantize_per_tensor.default(
+            quantized, scale, zero_point, low, high, dtype
+        )
+        return dequantized
+
+    def replacement_graph_per_tensor(weight, scale, zero_point, low, high, dtype):
+        return torch.ops.aten.mul.Tensor(weight, scale)
+
+    return pattern_per_tensor, replacement_graph_per_tensor
+
+
+def _set_meta_for_matches(model: torch.fx.GraphModule, matches: torch.fx.subgraph_rewriter.ReplacedPatterns) -> None:
+    """
+    Sets the meta for the nodes which were matched by the subgraph_rewriter pattern.
+    :param model: Model whose nodes' meta are to be set.
+    :param matches: matches returned by the subgraph_rewriter
+    """
+    for match in matches:
+        mul_node = match.replacements[0]
+        _set_new_node_meta(mul_node, list(mul_node.args), torch.mul, model)
+
+
+def _remove_constant_qdq_transformation(model: torch.fx.GraphModule) -> None:
+    """
+    Removes the Quantize/De-Quantize nodes for weight nodes by matching the pattern to be like follows:
+    weight scale zp
+        \   |    /
+         Quantize
+            |
+        Dequantize
+
+    and replace with the following:
+     weight scale
+       \     /
+       Multiply
+
+    :param model: Model to apply transformations to.
+    """
+
+    def match_filters(match, original_graph, graph):
+        for node in match.nodes_map:
+            if node.name == "weight" and match.nodes_map[node].op == "get_attr":
+                return True
+        return False
+
+    pattern, replacement = _get_pattern_replacement_per_channel()
+    matches_per_channel = torch.fx.subgraph_rewriter.replace_pattern_with_filters(
+        model, pattern, replacement, [match_filters]
+    )
+    _set_meta_for_matches(model, matches_per_channel)
+    pattern, replacement = _get_pattern_replacement_per_tensor()
+    matches_per_tensor = torch.fx.subgraph_rewriter.replace_pattern_with_filters(
+        model, pattern, replacement, [match_filters]
+    )
+    _set_meta_for_matches(model.graph, matches_per_tensor)
+
+    return matches_per_channel, matches_per_tensor
+
+
+def _get_node_inputs(node: torch.fx.Node, model: torch.fx.GraphModule) -> Optional[Tuple[Union[torch.Tensor, int]]]:
+    """
+    Gets the inputs for the Quantize node which quantize the weights. Otherwise return None
+    :param node: Node to get the inputs from
+    :param model: Model to which the node belongs
+    """
+    input_tup = []
```

Comment:
What do you think about renaming `input_tup` to `args` here and elsewhere?

Reply:
I assumed it could cause some confusion with node.args or other args variables too. Although I think it will be better since it returns the args but with tensors. I will change it

Reply:
Done